langchain/retrievers/perfect_retriever.py

```diff
@@ -0,0 +1,220 @@
+import json
+from abc import abstractmethod
+from typing import Any, Dict, List
+
+from langchain import LLMChain, PromptTemplate
+from langchain.agents import AgentExecutor
+from langchain.agents.agent_toolkits import VectorStoreInfo
+from langchain.llms import BaseLLM
+from langchain.schema import BaseOutputParser, BaseRetriever, Document
+from langchain.vectorstores.base import VectorStoreRetriever
+
+pinecone_format_instructions = """RESPONSE FORMAT
+----------------------------
+
+When responding to me please, please output a response in this format. Use a markdown code snippet formatted in the following schema:
+
+```json
+{{{{
+    "search_string": string \\ The string to compare to document contents,
+    "metadata_filter": {{{{
+        "<MetadataField>": {{{{
+            <Operator>: <Value>
+        }}}},
+        "<MetadataField>": {{{{
+            <Operator>: <Value>
+        }}}},
+    }}}} \\ The conditions on which to filter the metadata
+}}}}
+```
+
+If no metadata filters are needed you can leave "metadata_filter" as an empty map.
+
+Filtering metadata supports the following operators:
+    $eq - Equal to (number, string, boolean)
```

Comment:
* What is the reason for this kind of syntax for the operators?
* Do we want the ability to disable/enable operators based on what the underlying store supports?

Reply:
yea this is all based on pinecone atm https://docs.pinecone.io/docs/metadata-filtering

Reply:
if we wanted we could try creating our own syntax and then translators from our syntax to any vector stores syntax, but idk that that's really any simpler than just having separate prompts for each vectorstore